### PR TITLE
Bug 1216520 - InputTypes only supports switching when there is more than one layout enabled

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1bfdbeac743febf1df66f90839ef13186728d5de

File: apps/system/test/unit/input_layouts_test.js [hunk 1]
```diff
@@ -176,6 +177,37 @@ suite('InputLayouts', function() {
...
+      'number': []
+    };
+
+    var oldGroupToTypeTable = inputLayouts._groupToTypeTable;
+    inputLayouts._groupToTypeTable = {
+      'text': ['text', 'textarea'],
+      'password': ['password']
+    };
+
+    var realMozInputMethod = navigator.mozInputMethod;
```
Comment (albertopq) [inline apps/system/test/unit/input_layouts_test.js]: Shouldn't this be in the setup, in a way that can be used by all the test?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=63ea33d047344868dda31725e096ea658b1c8e79

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62a2fe7745b013ff552dbbdee3317318c37a70cf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62a2fe7745b013ff552dbbdee3317318c37a70cf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=15047f410ceb5f01787984a70d4931ac8120265c

**Pull request merged**